aws_kms_pkcs11.cpp

```diff
@@ -206,6 +206,22 @@ CK_RV C_Initialize(CK_VOID_PTR pInitArgs) {
 #ifdef AWS_SDK_USE_SYSTEM_PROXY
         awsConfig.allowSystemProxy = true;
 #endif
+        const char* endpoint = std::getenv("AWS_ENDPOINT_URL");
```

Comment:
The PR looks good, but this section to determine the endpoint looks fully copy/pasted each time below. Could you factor it into a util method and reuse that instead?

Reply:
Hello.. Do you have an example of one of those? If not, in which file should I put that new function?

Reply:
I'd create a new file (util.h and util.cpp). You can look at the debug.h and debug.ccp files as a starting point, but I think util.h could have just

```
Aws::Client::ClientConfiguration create_aws_config(std::string region);
```

Reply:
Hello,

I just pushed a new version with that option. Kindly ask for the review.

Reply:
Looks great! Thanks for making those changes. I've merged it.